tests/foreman/api/test_repository.py

```diff
@@ -60,53 +61,76 @@
 from robottelo.datafactory import invalid_http_credentials
 from robottelo.datafactory import invalid_names_list
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
 from robottelo.datafactory import valid_docker_repository_names
 from robottelo.datafactory import valid_http_credentials
 from robottelo.datafactory import valid_labels_list
 from robottelo.decorators import skip_if_not_set
-from robottelo.decorators.host import skip_if_os
 from robottelo.helpers import get_data_file
 from robottelo.helpers import read_data_file
-from robottelo.test import APITestCase
 
 
-class RepositoryTestCase(APITestCase):
-    """Tests for ``katello/api/v2/repositories``."""
+@pytest.fixture
+def repo_options(request, module_org, module_product):
+    """Return the options that were passed as indirect parameters."""
+    options = getattr(request, 'param', {}).copy()
+    options['organization'] = module_org
+    options['product'] = module_product
+    return options
+
+
+@pytest.fixture
+def env(module_org):
+    return entities.Environment(organization=[module_org]).create()
+
+
+@pytest.fixture
+def repo(repo_options):
+    """create a new repository."""
+    return entities.Repository(**repo_options).create()
 
-    @classmethod
-    def setUpClass(cls):
-        """Create an organization and product which can be re-used in tests."""
-        super().setUpClass()
-        cls.org = entities.Organization().create()
-        cls.product = entities.Product(organization=cls.org).create()
-        cls.http_proxy = entities.HTTPProxy(
-            name=gen_string('alpha', 15),
-            url=settings.http_proxy.auth_proxy_url,
-            username=settings.http_proxy.username,
-            password=settings.http_proxy.password,
-            organization=[cls.org.id],
-        ).create()
+
+@pytest.fixture
+def http_proxy(module_org):
+    """Create a new HTTP proxy."""
+    return entities.HTTPProxy(
+        name=gen_string('alpha', 15),
+        url=settings.http_proxy.auth_proxy_url,
+        username=settings.http_proxy.username,
+        password=settings.http_proxy.password,
+        organization=[module_org.id],
+    ).create()
+
+
+class TestRepository:
```

Comment:
There are nine different classes in this module, several with identically-named tests, and some with pytest markers applied at the class level. Removing the class names would also require renaming a lot of tests to clarify what type of repository they are for, e.g., ` TestDRPMRepository::test_positive_sync_publish_cv` -> `test_positive_sync_drpm_publish_cv`. Also, I haven't heard any good reasons for removing classes in the first place.

Reply:
I agree with that

Reply:
@tpapaioa in this case, it would make more sense to keep them, as there are duplicate names. The classes, in this case, are only there for classification, instead of functionality.
The reason why classes are no longer the default is that, in most cases, they aren't needed.